### PR TITLE
Add phase layer to qudit MMD loss function

### PR DESCRIPTION
**Pull request by willjmax**
This PR addressed bug #10031 by adding the phase layer to the qudit circuit config in the MMD loss function.

### CODE REVIEW

**Review by austingmhuang — APPROVED** [review]

File: pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py [hunk 1]
```diff
@@ -973,3 +980,106 @@ def test_raises_n_samples_le_one(self):
...
+        assert float(without) != float(with_phase)
+
+    def test_phase_layer_matches_exact_mmd(self):
+        """The estimator with a phase layer must be unbiased w.r.t. exact MMD^2."""
+        gates = {0: [[1, 0]], 1: [[0, 1]], 2: [[1, 1]]}
+        params = [0.37, 0.95, 0.73]
+        n_qudits, d, bandwidth, graph_type = 2, 3, 0.5, "cycle"
+
+        rng = np.random.default_rng(42)
+        X = rng.integers(0, d, (80, n_qudits))
```
Comment (soranjh) [inline pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py]: Please use longer descriptive names instead of single letters.

File: pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py [hunk 2]
```diff
@@ -973,3 +980,106 @@ def test_raises_n_samples_le_one(self):
...
+            n_qudits=n_qudits,
+            gates=gates,
+            n_samples=self.N_SAMPLES,
+            key=jax.random.PRNGKey(0),
+            phase_fn=self._phase_fn,
+        )
+        mmd_cfg = QuditMMDConfig(bandwidth=bandwidth, n_ops=self.N_OPS, graph_type=graph_type)
+        loss_fn = build_qudit_mmd_loss(config, mmd_cfg)
+
+        X_jnp = jnp.array(X)
```
Comment (soranjh) [inline pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py]: Same comment as above.

File: pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py [hunk 3]
```diff
@@ -973,3 +980,106 @@ def test_raises_n_samples_le_one(self):
...
+            idx = jax.random.choice(sample_key, len(X), shape=(40,), replace=False)
+            estimates.append(float(loss_fn(params_jnp, X_jnp[idx], key=loss_key)))
+
+        estimates = np.array(estimates)
+        mean_est = np.mean(estimates)
+        se = np.std(estimates, ddof=1) / np.sqrt(self.N_TRIALS)
+        z = abs(exact - mean_est) / se if se > 1e-15 else 0.0
+        assert (
+            z < self.Z_THRESHOLD
+        ), f"Z-test FAILED: z={z:.2f}, exact={exact:.6f}, mean={mean_est:.6f}, se={se:.6f}"
```
Comment (soranjh) [inline pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py]: Why this is needed?

File: pennylane/labs/tcdq/qudit_mmd_loss.py
Comment (soranjh) [inline pennylane/labs/tcdq/qudit_mmd_loss.py:1]: Please update the changelog.

File: pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py (hunk 3 above)
Comment (willjmax) [reply pennylane/labs/tests/tcdq/test_qudit_mmd_loss.py]: This test uses a statistical Z test to determine if the estimator is unbiased. We compute the estimates of the `mmd_loss` function across multiple trials, then compare them with the true deterministic value that we precomputed with the call to `_exact_qudit_mmd2_kernel`. If the Z score is high enough we can say with high confidence that the estimator is unbiased.

**Review by soranjh — APPROVED** [review]
